EiDiA_backend/src/controllers/record.js

```diff
@@ -59,6 +72,129 @@ const addRecord = (req, res) => {
+    DocumentModel.aggregate([{$match: {recordId: mongoose.Types.ObjectId(req.params.recordId)}},
+            {$group: {_id: '$documentTypeId'}},
+            {$lookup: {from: 'documenttypes', localField: '_id', foreignField: '_id', as: 'test'}},
+            {$unwind: "$test"},
+            {
+                "$project": {
+                    "_id": 1,
+                    "docTypeId": "$_id",
+                    "name": "$test.name"
+                }
+            }
+        ],
+        function (err, documents) {
+            //TODO: check for error
+            res.status(200).json({documents: documents});
+
+        });
+
+}
+const getAttributeTypesForRecord = (req, res) => {
+    //TODO: add sets for unique values
+    DocumentModel.aggregate([{$match: {recordId: mongoose.Types.ObjectId(req.params.recordId)}},
+            {$group: {_id: '$documentTypeId', attributes: {$first: "$attributes.attributeId"}}},
+            {$lookup: {from: 'attributetypes', localField: 'attributes', foreignField: '_id', as: 'test'}},
+            {$unwind: "$test"},
+            {
+                "$project": {
+                    "_id": 1,
+                    "docTypeId": "$_id",
+                    "attributeId": "$test._id",
+                    "name": "$test.name"
+                }
+            }
+        ],
+        function (err, documents) {
+            //TODO: check for error
+
+            console.log(documents);
+            res.status(200).json(documents);
+
+        });
+
+}
+const getAttributeValuesForRecord = (req, res) => {
+    //TODO: add sets for unique values
+    console.log("hello")
+    DocumentModel.aggregate([{$match: {recordId: mongoose.Types.ObjectId(req.params.recordId)}},
+            {$unwind: "$attributes"},
+
+            {
+                "$project": {
+
+                    "docTypeId": "$documentTypeId",
+                    "date": "$createdOnDate",
+                    "attributeId": "$attributes.attributeId",
+                    "value": "$attributes.value",
+
+                }
+            },
+
+        ],
+        function (err, documents) {
+            //TODO: check for error
+
+            res.status(200).json(documents);
+
+        });
+
+}
+
+const listDocumentByRecordId = (req, res) => { // Return attributes based on selected DocumentTypeId
+
+
+    DocumentModel.find({'recordId': mongoose.Types.ObjectId(req.params.recordId)})
+        .then(documentList => {
+            let response = documentList.map(document => {
+                return {
+                    id: document._id,
+                    name: document.name,
+                    parentFolderId: document.parentFolderId.toString() === "000000000000000000000000" ? 0 : document.parentFolderId,
+                    fileType: document.fileType, //TODO: implement filetype in database
+                    createdOnDate: document.createdOnDate,
+                    lastModifiedOnDate: document.lastModifiedOnDate,
```

Comment:
```suggestion
                    dateCreation: format(document.createdOnDate, 'dd/MM/yyyy'),
                    dateModification: format(document.lastModifiedOnDate, 'dd/MM/yyyy'),
```
dann musst du im Frontend nicht convertieren